stac_model/output.py

```diff
@@ -0,0 +1,43 @@
+from enum import Enum
+from typing import List, Optional, Union
+
+from pydantic import BaseModel, Field
+
+
+class TaskEnum(str, Enum):
+    regression = "regression"
+    classification = "classification"
+    object_detection = "object detection"
+    semantic_segmentation = "semantic segmentation"
+    instance_segmentation = "instance segmentation"
+    panoptic_segmentation = "panoptic segmentation"
+    multi_modal = "multi-modal"
+    similarity_search = "similarity search"
+    image_captioning = "image captioning"
+    generative = "generative"
+    super_resolution = "super resolution"
+
+
+class ResultArray(BaseModel):
+    shape: List[Union[int, float]]
+    dim_names: List[str]
+    data_type: str = Field(
+        ...,
+        pattern="^(uint8|uint16|uint32|uint64|int8|int16|int32|int64|float16|float32|float64)$",
+    )
+
+
+class ClassObject(BaseModel):
+    value: int
+    name: str
+    description: Optional[str] = None
+    title: Optional[str] = None
+    color_hint: Optional[str] = None
+    nodata: Optional[bool] = False
```

Comment:
gotcha yeah I wasn't familiar with how to combine pystac and pydantic validation so I initially leaned toward doing all of the validation and serialization with pydantic sinc eit seemed simpler and not too much effort.

But from talking with you it sounds like we should reuse the pystac classes and extend a pystac item, letting pystac handle the serialization and deserialization and pydantic only handle our extension' property validation? I'm all for the reuse, just a bit shaky on how we get to validating all of these extension properties neatly

Reply:
Indeed. If we find a neat way to make validation work with either a `pydantic` or `pystac` validation approach interchangeably, that would be the best. However, we must expect that users creating STAC Items/Collections will primarily interact with `pystac` since this is the current tool available for it. Under the hood, it seems acceptable to use `pydanctic` models if that feels easier to implement.

Reply:
see https://github.com/crim-ca/dlm-extension/pull/2#discussion_r1517087799 for a potential solution for these pydantic/pystac interop issues